src/db/gc/mod.rs

```diff
@@ -204,6 +223,8 @@ impl<DB: Blockstore + GarbageCollectable + Sync + Send + 'static> MarkAndSweep<D
         info!("GC sweep");
         self.sweep()?;
 
+        self.update_last_gc_run(current_epoch)?;
```

Comment:
I'm wondering why it has to be `current_epoch` instead of `current_epoch - 2 * CHAIN_FINALITY`. That way, we don't have to wait

Reply:
The original idea was that we don’t want to continuously run this, because the graph traversal takes up significant amount of resources. And there is no real reason to do so, as we don’t get so much unreachable stuff on each run.

Reply:
This idea is to avoid waiting during a GC run. We can still set threshold between multiple runs to reduce frequency. Does that make sense or did I misunderstand something?

Reply:
We discussed that before - there is no way for us to avoid waiting during a GC run, because the only way to know the age of any given CID is by knowing when we marked it, as we are iterating all the available keys, without knowing which epoch they belong to. The GC makes no assumptions about the marked CIDs. 

This PR fixes calibnet GC, feel free to propose design improvements via an issue, then we can review those and see what’s possible.

Reply:
About the last run epoch - we record the epoch GC fully executed last. The way that it’s further used in calculations for the next run is whatever we decide it to be. It’s more logical to use the actual epoch when the GC finished as opposed to some calculated value.

Reply:
As I understand, the algorithm needs to wait for `depth` epochs, but logically, these epochs don't have to be from the future, and what I suggest is to wait for existing `depth` epochs in the same algorithm, to avoid waiting in time, does that make sense?

Reply:
> As I understand, the algorithm needs to wait for `depth` epochs, but logically, these epochs don't have to be from the future, and what I suggest is to wait for `depth` epochs that already exist in the database with the same algorithm to avoid waiting in time (`EPOCH_DELAY * depth`), does that make sense?

It doesn't make sense, because we populate keys based on iterating everything that exists in the database (ParityDB, not CarDB). So we have absolutely no idea which epoch the CID belongs to. This is why we have to wait `2*chain_finality` after we have marked certain keys for deletion.